modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/DiscoverFragment.kt

```diff
@@ -260,28 +266,29 @@ class DiscoverFragment : BaseFragment(), DiscoverAdapter.Listener, RegionSelectF
         return mutableContentList
     }
 
-    private fun updateDiscover(mostPopularPodcastsByCategoryRow: MostPopularPodcastsByCategoryRow) {
+    private fun updateDiscover(
+        mostPopularPodcasts: MostPopularPodcastsByCategoryRow,
+        remainingPodcasts: RemainingPodcastsByCategoryRow,
+    ) {
         adapter?.currentList?.let { discoverList ->
-            val updatedList =
-                discoverList.filterNot { it is DiscoverRow && it.id == "featured" }.toMutableList() // Remove ads carousel
+            val updatedList = discoverList.filter { it is DiscoverRow && it.type is ListType.Categories }.toMutableList()
 
-            updatedList.add(MOST_POPULAR_PODCASTS_ROW_INDEX, mostPopularPodcastsByCategoryRow)
+            updatedList.add(MOST_POPULAR_PODCASTS_ROW_INDEX, mostPopularPodcasts)
+            updatedList.add(REMAINING_PODCASTS_ROW_INDEX, remainingPodcasts)
 
             adapter?.submitList(updatedList)
         }
     }
-    private fun clearCategoryFilter() {
-        adapter?.currentList?.toMutableList()?.apply {
-            removeAll { it is MostPopularPodcastsByCategoryRow }
-            adapter?.submitList(this)
-            viewModel.loadData(resources)
-        }
+    private fun reloadDiscover() {
+        viewModel.onCleared()
```

Comment:
This shouldn't ever be called by us. It is a system callback that is managed by the view model and using it can lead to hard to discover bugs in the future. Imagine that we would add some long running operations to the view model. Or have forever running jobs during initialization. This would cancel them. If you need to clear some resources be explicit about it here.